flac.md

```diff
@@ -544,61 +544,28 @@ Data      | Description
-`RESIDUAL_CODING_METHOD_PARTITIONED_EXP_GOLOMB` \|\| `RESIDUAL_CODING_METHOD_PARTITIONED_EXP_GOLOMB2` |
+## Coded residual
+The first two bits in a coded residual indicate which coding method is used. See the table below
 
-### RESIDUAL_CODING_METHOD
 Value       | Description
 -----------:|:-----------
-0b00        | partitioned Exp-Golomb coding with 4-bit Exp-Golomb parameter; RESIDUAL_CODING_METHOD_PARTITIONED_EXP_GOLOMB follows
-0b01        | partitioned Exp-Golomb coding with 5-bit Exp-Golomb parameter; RESIDUAL_CODING_METHOD_PARTITIONED_EXP_GOLOMB2 follows
+0b00        | partitioned Rice code with 4-bit parameters
+0b01        | partitioned Rice code with 5-bit parameters
 0b10 - 0b11 | reserved
 
-### RESIDUAL_CODING_METHOD_PARTITIONED_EXP_GOLOMB
-Data              | Description
-:-----------------|:-----------
-`u(4)`            | Partition order.
-`EXP_GOLOMB_PARTITION`+ | There will be 2\^order partitions.
+Both defined coding methods work the same way, but differ in the number of bits used for rice parameters. The 4 bits that directly follow the coding method bits form the partition order, which is an unsigned number. The rest of the coded residual consists of 2^(partition order) partitions. For example, if the 4 bits are 0b1000, the partition order is 8 and the residual is split up into 2^8 = 256 partitions.
 
-#### EXP_GOLOMB_PARTITION
-Data       | Description
-:----------|:-----------
-`u(4(+5))` | `EXP-GOLOMB PARTITION ENCODING PARAMETER` (see [section on EXP-GOLOMB PARTITION ENCODING PARAMETER](#exp-golomb-partition-encoding-parameter))
-`u(?)`     | `ENCODED RESIDUAL` (see [section on ENCODED RESIDUAL](#encoded-residual))
+Each partition contains a certain amount of residual samples. The number of residual samples in the first partition is equal to (blocksize >> partition order) - predictor order, i.e. the blocksize divided by the number of partitions minus the predictor order. In all other partitions the number of residual samples is equal to (blocksize >> partition order).
 
-#### EXP GOLOMB PARTITION ENCODING PARAMETER
-Value           | Description
----------------:|:-----------
-0b0000 - 0b1110 | Exp-golomb parameter.
-0b1111          | Escape code, meaning the partition is in unencoded binary form using n bits per sample; n follows as a 5-bit number.
-
-### RESIDUAL_CODING_METHOD_PARTITIONED_EXP_GOLOMB2
-Data               | Description
-:------------------|:-----------
-`u(4)`             | Partition order.
-`EXP-GOLOMB2_PARTITION`+ | There will be 2\^order partitions.
+The partition order MUST be so that the blocksize is evenly divisible by the number of partitions. This means for example that for all uneven blocksizes, only partition order 0 is allowed.  The partition order also MUST be so that the (blocksize >> partition order) is larger than the predictor order. This means for example that with a blocksize of 4096 and a predictor order of 4, partition order cannot be larger than 9.
```

Comment:
```suggestion
The partition order MUST be so that the blocksize is evenly divisible by the number of partitions. This means for example that for all odd blocksizes, only partition order 0 is allowed.  The partition order also MUST be so that the (blocksize >> partition order) is larger than the predictor order. This means for example that with a blocksize of 4096 and a predictor order of 4, partition order cannot be larger than 9.
```